### PR TITLE
JCRVLT-662: JcrPackageRegistry.remove() leaks temp file

### PR DESCRIPTION
**Pull request by mreutegg**
Fix and test to validate

### CODE REVIEW

File: vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java [hunk 1]
```diff
@@ -655,8 +655,8 @@ public void remove(@NotNull PackageId id) throws IOException, NoSuchPackageExcep
         if (pkg == null) {
             throw new NoSuchPackageException().setId(id);
         }
-        JcrPackage pack = pkg.getJcrPackage();
         try {
+            JcrPackage pack = pkg.getJcrPackage();
```
Comment (kwin) [inline vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java:659]: you can use [try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) as `JcrPackage` implements `AutoClosable`.

Comment (mreutegg) [reply vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java:659]: Ah, missed that one. Though, it seems it is not necessary in this case. Closing `JcrRegisteredPackage` already closes the `JcrPackage` it holds.

Comment (mreutegg) [reply vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java]: Done. Also put the `snap` `JcrPackage` into a try with resources.

**Review by kwin — APPROVED** [review]
LGTM

**Pull request merged**